src/server-glue/webresource-handler.ts

```diff
@@ -0,0 +1,327 @@
+		return false;
+	}
+
+	const apiRoot = getApiRoot(req);
+	if (apiRoot == null) {
+		return false;
+	}
+	const model = getModel(apiRoot);
+	const { resourceName } = await uriParser.parseOData({
+		url: req.url,
+		method: req.method,
+	});
+
+	const permission = req.method === 'POST' ? 'create' : 'update';
+	const vocab = model.versions[model.versions.length - 1];
+	const hasPermissions = await checkPermissions(
+		req,
+		permission,
+		resourceName,
+		vocab,
+	);
+
+	if (!hasPermissions) {
+		return false;
+	}
+
+	// TODO: This could be cached
+	const fields = model.abstractSql.tables[resourceName].fields;
+	const dbFieldName = odataNameToSqlName(fieldname);
+	for (const field of fields) {
+		if (field.fieldName === dbFieldName && field.dataType === 'WebResource') {
+			return true;
+		}
+	}
+
+	// TODO: We could do a pre-check if there is a SBVR rule specifying file max size
+	// This would avoid needing the roundtrip to DB and uploading a file just to remove it
+
+	return false;
+};
+
+export const getUploaderMiddlware = (
+	handler: WebResourceHandler,
+): Express.RequestHandler => {
+	return async (req, res, next) => {
+		if (!is(req, ['multipart'])) {
+			return next();
+		}
+		const filesUploaded: string[] = [];
+		const completeUploads: Array<Promise<void>> = [];
+
+		const bb = busboy({ headers: req.headers });
+		let isAborting = false;
+
+		const done = () => {
+			req.unpipe(bb);
+			req.on('readable', req.read.bind(req));
+			bb.removeAllListeners();
+		};
+
+		const clearFiles = () => {
+			isAborting = true;
+			const deletions = filesUploaded.map((file) => handler.removeFile(file));
+			// Best effort: We try to remove all uploaded files, but if this fails, there is not much to do
+			return Promise.all(deletions).catch((err) =>
+				console.error('Error deleting file', err),
+			);
+		};
+
+		bb.on('file', async (fieldname, filestream, info) => {
+			if (!isAborting && (await ifFileInValidPath(fieldname, req))) {
+				const file: IncomingFile = {
+					originalname: info.filename,
+					encoding: info.encoding,
+					mimetype: info.mimeType,
+					stream: filestream,
+					fieldname,
+				};
+				const promise = handler.handleFile(file).then((result) => {
+					req.body[fieldname] = {
+						filename: info.filename,
+						contentType: info.mimeType,
+						contentDisposition: undefined,
+						size: result.size,
+						href: result.filename,
+					};
+					filesUploaded.push(result.filename);
+				});
+				completeUploads.push(promise);
+			} else {
+				filestream.resume();
+			}
+		});
+
+		bb.on('field', (name, val, _info) => {
+			req.body[name] = val;
+		});
+
+		bb.on('finish', async () => {
+			try {
+				await Promise.all(completeUploads);
+				done();
+				next();
+			} catch (err: any) {
+				await clearFiles();
+
+				if (err?.message.includes('File size exceeded')) {
+					return sbvrUtils.handleHttpErrors(
+						req,
+						res,
+						new errors.BadRequestError('File size exceeded'),
+					);
+				}
```

Comment:
Or more specific WebResourceError

Reply:
@fisehara  I guess FileSizeExceededError no? Because that's the only case where we want to throw the 400. The others are really 500. Maybe we can have both where FileSizeExceededError extends WebResourceError

Reply:
@otaviojacobi that sounds appropriate, yes!